core/src/test/java/com/graphhopper/routing/util/spatialrules/SpatialRuleLookupJTSTest.java

```diff
@@ -130,20 +147,45 @@ public void testExactAdjacentBorder() {
     @Test
     public void testSmallBoundsBigPolygon() {
         List<SpatialRule> spatialRules = new ArrayList<>();
-        spatialRules.add(getSpatialRule(new Polygon(new double[]{-100, -100, 100, 100}, new double[]{-100, 100, 100, -100}), "big"));
-        SpatialRuleLookup spatialRuleLookup = new SpatialRuleLookupArray(spatialRules, 1, true, new BBox(1, 2, 1, 2));
+        Polygon big = FAC.createPolygon(new Coordinate[] { new Coordinate(-100, -100), new Coordinate(100, -100), new Coordinate(100, 100), new Coordinate(-100, 100), new Coordinate(-100, -100)});
+        spatialRules.add(getSpatialRule(big, "big"));
+        SpatialRuleLookup spatialRuleLookup = new SpatialRuleLookupJTS(spatialRules, new Envelope(1, 2, 1, 2));
         assertEquals("big", spatialRuleLookup.lookupRule(1.5, 1.5).getId());
     }
+    
+    @Test
+    public void testHole() {
```

Comment:
It's great that you added this test!

I was just wondering, why is the hole the same as the shell?

Wouldn't it make more sense to have an actual polygon with a hole and a border like:

```
LinearRing shell = FAC.createLinearRing(new Coordinate[] { new Coordinate(1, 1), new Coordinate(7, 1), new Coordinate(7, 7), new Coordinate(1, 7), new Coordinate(1, 1)});
LinearRing hole = FAC.createLinearRing(new Coordinate[] { new Coordinate(2, 2), new Coordinate(6, 2), new Coordinate(6, 6), new Coordinate(2, 6), new Coordinate(2, 2)});
```

And then we could test like:

```
assertEquals("SpatialRule.EMPTY", spatialRuleLookup.lookupRule(5, 3).getId());
assertEquals("1", spatialRuleLookup.lookupRule(1.5, 1.5).getId());
```

Then I think we could drop the p2 part of the test?

Reply:
This was the sole reason that the test passed. After i switched it to the intended hole structure the test failed because lookupRule() wants lat/lon while all JTS code is lon/lat....

Reply:
Sorry, I am not sure if I understand your reasoning here, could you try to explain this a bit more? My idea was that it's probably rather uncommon to have a polygon that is a complete hole?

Reply:
The hole was not intended to cover the whole polygon. I think that i messed up a rebase at some point. After fixing it i discovered that the test doesn't pass anymore because i mixed up lat/lon in the lookup. Long story short: two bugs squashed :smile:

Reply:
Ah I see, thanks for the explanation :)

Reply:
Sidenote: We should probably settle on one representation for coordinate values because GH code seems to prefer lat/lon while JTS is using lon/lat.

Reply:
Probably it would be better to stay with lat,lon in order to keep the existing methods/interfaces working as they are? Since this would only switch two doubles, it would be hard for external parties to see the change when merging.

Reply:
Yeah, you're right. I think it's fine as long as we keep the contact area between these two coordinate representations low.